In [1]:
# Import library
import pandas as pd

# Initial Cleaning

In [29]:
# Load dataset
teaser = pd.read_csv('./data/teaser_comments.csv')
debut = pd. read_csv('./data/debut_comments.csv')

### Data Exploration:

In [30]:
teaser.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 33086 entries, 0 to 33085
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   group        33086 non-null  object
 1   video_id     33086 non-null  object
 2   publishedAt  33086 non-null  object
 3   author       33086 non-null  object
 4   text         33086 non-null  object
 5   likeCount    33086 non-null  int64 
dtypes: int64(1), object(5)
memory usage: 1.5+ MB


In [31]:
debut.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 174589 entries, 0 to 174588
Data columns (total 6 columns):
 #   Column       Non-Null Count   Dtype 
---  ------       --------------   ----- 
 0   group        174589 non-null  object
 1   video_id     174589 non-null  object
 2   publishedAt  174589 non-null  object
 3   author       174589 non-null  object
 4   text         174578 non-null  object
 5   likeCount    174589 non-null  int64 
dtypes: int64(1), object(5)
memory usage: 8.0+ MB


In [32]:
# Convert likeCount to int
teaser["likeCount"] = teaser["likeCount"].astype(int)
debut["likeCount"] = debut["likeCount"].astype(int)

After dropping relevant NaNs, our dataset has a total of 33,086 comments from teaser videos. I notice that there are 29,049 unique comments by 24,011 unqiue authors. This tells us that there are some authors that may post the same comment multiple times. To account for this, we will remove dpulicate comments that are written by the same author for the same video.

In [ ]:
# drop duplicates for rows that have the same video id, author, and text
teaser = teaser.drop_duplicates(subset=["video_id", "author", "text"], keep="first")
debut = debut.drop_duplicates(subset=["video_id", "author", "text"], keep="first")

# There is nas in debut comments- Specifically text variable
debut = debut.dropna()

print(teaser.info())
print(debut.info())


<class 'pandas.core.frame.DataFrame'>
Index: 32705 entries, 0 to 33085
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   group        32705 non-null  object
 1   video_id     32705 non-null  object
 2   publishedAt  32705 non-null  object
 3   author       32705 non-null  object
 4   text         32705 non-null  object
 5   likeCount    32705 non-null  int64 
dtypes: int64(1), object(5)
memory usage: 1.7+ MB
None
<class 'pandas.core.frame.DataFrame'>
Index: 174578 entries, 0 to 174588
Data columns (total 6 columns):
 #   Column       Non-Null Count   Dtype 
---  ------       --------------   ----- 
 0   group        174578 non-null  object
 1   video_id     174578 non-null  object
 2   publishedAt  174578 non-null  object
 3   author       174578 non-null  object
 4   text         174578 non-null  object
 5   likeCount    174578 non-null  int64 
dtypes: int64(1), object(5)
memory usage: 9.3+ MB
None


Now, we've removed 381 duplicates for comments that have the same content, written by the same author, and for the same video. That should put us a total of 32,705 teaser comments.

In [34]:
len(teaser)

32705

In [41]:
# read the cleaned data into a new csv
teaser.to_csv('./data/teaser_comments.csv', index = False)
debut.to_csv('./data/debut_comments.csv', index = False)

# Basic Statistical Analysis

In [36]:
# comment counts per group
group_counts = teaser["group"].value_counts()
print(group_counts)

group
Kep1er         11318
ZEROBASEONE     7338
NMIXX           5600
LE SSERAFIM     2863
IVE             2723
TNX             1178
RIIZE           1050
BOYNEXTDOOR      362
H1-KEY           152
YOUNITE          121
Name: count, dtype: int64


I notice that groups that are more popular generally have higher comment counts

In [37]:
# most liked comments
top_comments = teaser.sort_values("likeCount", ascending=False).head(10)
print(top_comments[["group", "likeCount", "text"]])

             group  likeCount  \
16362       Kep1er      26548   
7591         NMIXX       9742   
20971       Kep1er       7202   
6396         NMIXX       6935   
2489   LE SSERAFIM       6652   
5318           IVE       5753   
9352         NMIXX       5508   
6243         NMIXX       5066   
5630           IVE       4938   
1405   LE SSERAFIM       3890   

                                                    text  
16362                         They are finally debuting😭  
7591              OH MY FREAKY FISHY! THEY LOOK SO GOOD!  
20971  ALL THE BEST KEP1ER QUEENS ROCK IT MY FULL SUP...  
6396   Wait, is this an entirely different song? Are ...  
2489   It&#39;s like saying &quot;why is it too short...  
5318   Their debut is getting nearer and nearer, It&#...  
9352            It will be iconic, I CAN&#39;T WAITTT 🔥🔥  
6243   QUE TEASER FOI ESSE ??? JYP Finalmente sabendo...  
5630                        YOU MAKE ME FEEL LIKE ELEVEN  
1405   Obsessed with the fact that they are da

I notice that a majority of the comments with the most likes are positive, suggesting that it is likelt that the majority of teaser comments show positive sentiment. 

In [38]:
# most common words
from collections import Counter
words = " ".join(teaser["text"]).lower().split()
Counter(words).most_common(20)

[('the', 3746),
 ('i', 3554),
 ('so', 3540),
 ('is', 2606),
 ('this', 2442),
 ('a', 1999),
 ('omg', 1916),
 ('for', 1790),
 ('to', 1767),
 ('wait', 1587),
 ('can&#39;t', 1510),
 ('debut', 1352),
 ('and', 1352),
 ('be', 1280),
 ('of', 1218),
 ('my', 1212),
 ('it', 1145),
 ('love', 1133),
 ('excited', 1034),
 ('da', 997)]

I notice while there are many stopwords here as the most common words, there are some words that show positive sentiment such as 'omg', 'love', and 'excited', suggesting that positive sentiment may be more prevalent in the teaser videos than negative sentiment. 

In [4]:
data = pd.read_csv('./data/debut_sentimental_analysis_roberta.csv')

In [5]:
data

,group,video_id,publishedAt,author,text,likeCount,roberta_sentiment,roberta_confidence
0,BOYNEXTDOOR,jizAb-SLvtM,2023-05-25T20:58:41Z,@bichinabichina4452,mid,0,neutral,0.559092
1,BOYNEXTDOOR,jizAb-SLvtM,2023-05-25T20:57:41Z,@кирусядеточкина,"THEY&#39;RE SOOO SWEET I REALLY LOVE IT, GUYS ...",0,positive,0.980104
2,BOYNEXTDOOR,jizAb-SLvtM,2023-05-25T20:57:40Z,@Mama_Aqila01,Vibes nya prince bgt ga sih? Para prince yg la...,0,positive,0.919478
3,BOYNEXTDOOR,jizAb-SLvtM,2023-05-25T20:57:24Z,@leeknooooow,이 옵션을 선택하면 이 옵션을 선택할 수 있습니다♥️😍😅,1,positive,0.815676
4,BOYNEXTDOOR,jizAb-SLvtM,2023-05-25T20:57:08Z,@loove143_,0:14 &quot;representing KOZ&quot; I can&#39;t ...,146,positive,0.882579
...,...,...,...,...,...,...,...,...
174573,ZEROBASEONE,trzeUClQIIg,2023-07-10T09:00:44Z,@mymymaxine,OMG,0,positive,0.591295
174574,ZEROBASEONE,trzeUClQIIg,2023-07-10T09:00:44Z,@nozvia,FIRST,0,neutral,0.468150
174575,ZEROBASEONE,trzeUClQIIg,2023-07-10T09:00:43Z,@조용히해기무낙,0:01 아니 0:00 부터 헐 미쳤다 ㅠㅠㅠ 🎉🎉 제베원 데뷔 경축.. ㅠㅠ!!!...,1,neutral,0.510665
174576,ZEROBASEONE,trzeUClQIIg,2023-07-10T09:00:43Z,@Kyo02,♡,0,positive,0.830632
